src/components/AddProductStoreToGroupModal.vue

```diff
@@ -0,0 +1,185 @@
+  methods: {
+    closeModal() {
+      modalController.dismiss({ dismissed: true});
+    },
+    async fetchGroupProductStores() {
+       try {
+        const resp = await FacilityService.fetchAssociatedProductStoresToGroup({
+          "inputFields": {
+            "facilityGroupId": this.group.facilityGroupId
+          },
+          "viewSize": 250, // maximum view size
+          "entityName": 'ProductStoreFacilityGroup',
+          "noConditionFind": "Y",
+          "filterByDate": 'Y'
+        })
+
+        if(!hasError(resp)) {
+          this.selectedProductStores = resp.data.docs
+          this.selectedProductStoreValues = JSON.parse(JSON.stringify(resp.data.docs))
+        } else {
+          throw resp.data
+        }
+      } catch(err) {
+        logger.error(err)
+      }
+    },
+    isSelected(productStoreId: any) {
+      return this.selectedProductStoreValues.some((productStore: any) => productStore.productStoreId === productStoreId);
+    },
+    toggleProductStoreSelection(updatedStore: any) {
+      let selectedStore = this.selectedProductStoreValues.some((store: any) => store.productStoreId === updatedStore.productStoreId);
+
+      if(selectedStore) {
+        this.selectedProductStoreValues = this.selectedProductStoreValues.filter((store: any) => store.productStoreId !== updatedStore.productStoreId);
+      } else {
+        this.selectedProductStoreValues.push(updatedStore);
+      }
+    },
+    async saveProductStores() {
+      const productStoresToAdd = this.selectedProductStoreValues.filter((selectedStore: any) => !this.selectedProductStores.some((store: any) => store.productStoreId === selectedStore.productStoreId))
+      const productStoresToRemove = this.selectedProductStores.filter((store: any) => !this.selectedProductStoreValues.some((selectedStore: any) => store.productStoreId === selectedStore.productStoreId))
+
+      const removeResponses = await Promise.allSettled(productStoresToRemove
+        .map(async (store: any) => await FacilityService.updateProductStoreFacilityGroup({
+          "productStoreId": store.productStoreId,
+          "facilityGroupId": this.group.facilityGroupId,
+          "fromDate": store.fromDate,
+          "thruDate": DateTime.now().toMillis()
+        }))
+      )
+
+      const addResponses = await Promise.allSettled(productStoresToAdd
+        .map(async (store: any) => await FacilityService.createProductStoreFacilityGroup({
+          "productStoreId": store.productStoreId,
+          "facilityGroupId": this.group.facilityGroupId,
+          "fromDate": DateTime.now().toMillis()
+        }))
+      )
+
+      const hasFailedResponse = [...removeResponses, ...addResponses].some((response: any) => response.status === 'rejected')
+      if (hasFailedResponse) {
+        showToast(translate("Failed to associate some product stores to group."))
+      } else {
+        showToast(translate("Product stores associated to group successfully."))
+      }
+      this.fetchGroups()
```

Comment:
Do not fetch all the groups again, just update the count for currentGroup in state.